### ISSUE TITLE
withDefaultVariant typescript check variant expect string but got an undefined

### ISSUE THREAD

**Issue by Mayvis** [chakra-ui/chakra-ui#5440]
### Description

When use withDefaultVariant typescript check variant expect string but got an undefined

### Link to Reproduction

https://codesandbox.io/s/delicate-resonance-8hdcw?file=/src/theme.ts

### Steps to reproduce

1. Use Vite to quickly create react-ts project by running `yarn create vite my-vue-app --template react-ts`
2. Install chakra-ui by running `yarn add @chakra-ui/react @emotion/react@^11 @emotion/styled@^11 framer-motion@^5`
3. Create theme.js in src folder
```typescript
// theme.js
import { extendTheme, withDefaultVariant } from "@chakra-ui/react";

const theme = extendTheme(
  withDefaultVariant({
    variant: "filled", // with-default-variant.d.ts(5, 5): The expected type comes from property 'variant' which is declared here on type '{ variant: undefined; components?: Dict<any> | string[] | undefined; }'
    components: ["Input", "NumberInput", "Textarea"],
  })
);

export default theme;
```

### Chakra UI Version

1.7.4

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

![截圖 2022-01-24 下午5 07 39](https://user-images.githubusercontent.com/12667910/150753325-eaab6dc2-3222-42af-9c2a-b4197e56cbde.png)

**Comment by BigDog1400** [comment]
I have a similar bug but with a different error type
![image](https://user-images.githubusercontent.com/63280663/150886486-72ca4a8f-6c34-4700-8fc5-dbb26b94a574.png)
![image](https://user-images.githubusercontent.com/63280663/150886503-dc80d90e-b10c-4d4e-b05e-e97536a27c39.png)

**Comment by dca123** [comment]
I had to do 

- an exact revert to , `"@chakra-ui/react": "1.7.3",`, 
- clear yarn cache
- remove node_modules folder

to get this issue fixed.

I hope this is fixed as it's preventing builds or some documentation on how types can be added for variants.

**Comment by TimKolberger** [comment]
Kindly check if this issue persists with @chakra-ui/react@1.8.1.

**Comment by Mayvis** [comment]
@TimKolberger, issue had been solved when I upgrade to @chakra-ui/react@1.8.1. Thank you.🙂
![截圖 2022-01-27 上午11 17 39](https://user-images.githubusercontent.com/12667910/151285965-69505c64-796e-4612-a220-62b957bbf7aa.png)
